src/python/pants/backend/java/subsystems/javac.py

```diff
@@ -17,10 +19,19 @@ class JavacSubsystem(Subsystem):
     @classmethod
     def register_options(cls, register):
         super().register_options(register)
+        register(
+            "--args",
+            type=list,
+            member_type=shell_str,
+            default=[],
+            help=("Global `javac` compiler flags."),
```

Comment:
Whenever we use shell string, we tend to give examples to make clear how you don't have to worry about shlexing. Might be relevant to do here.